qiskit/execute.py

```diff
@@ -261,33 +262,44 @@ def execute(experiments, backend,
                                meas_map=meas_map,
                                method=scheduling_method)
 
-    # assembling the circuits into a qobj to be run on the backend
-    qobj = assemble(experiments,
-                    qobj_id=qobj_id,
-                    qobj_header=qobj_header,
-                    shots=shots,
-                    memory=memory,
-                    max_credits=max_credits,
-                    seed_simulator=seed_simulator,
-                    default_qubit_los=default_qubit_los,
-                    default_meas_los=default_meas_los,
-                    schedule_los=schedule_los,
-                    meas_level=meas_level,
-                    meas_return=meas_return,
-                    memory_slots=memory_slots,
-                    memory_slot_size=memory_slot_size,
-                    rep_time=rep_time,
-                    parameter_binds=parameter_binds,
-                    backend=backend,
-                    init_qubits=init_qubits,
-                    **run_config)
-
-    # executing the circuits on the backend and returning the job
-    start_time = time()
-    job = backend.run(qobj, **run_config)
-    end_time = time()
-    _log_submission_time(start_time, end_time)
-    return job
+    # v2 Providers Backend
+    if isinstance(backend, Backend):
+        start_time = time()
+        backend.set_configuration(shots=shots, **run_config)
```

Comment:
We do, it's part of the backwards compat/migration story. The v2 providers interface is not a user facing thing directly. It's provider api facing, a terra end user won't necessarily know that when they install a new version of aer or ibmq the api version the provider is talking to terra with has changed. `execute` in particular is very important because it has been the primary entrypoint we've documented to users for a long time so they'll have existing code leveraging it. I agree, the backwards compat story here isn't finished yet I only did enough to validate execute worked with basicaer. I'll also need to add a check if the configuration object for a v2 backend has the attribute before trying to pass it `set_configuration()`.